webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php

```diff
@@ -224,7 +224,16 @@ public function visitWithNoContest(string $url, bool $dropdown): void
         $response = $this->client->getResponse();
         self::assertNotEquals(500, $response->getStatusCode(), sprintf('Failed at %s', $url));
         if ($dropdown && !str_contains($url, '/public')) {
-            self::assertSelectorExists('a#navbarDropdownContests:contains("no contest")');
+            try {
+                self::assertSelectorExists('a#navbarDropdownContests:contains("no contest")', "Failed at: " . $url);
+            } catch (\Exception $e) {
+                $gitlabArtifacts = getenv('GITLABARTIFACTS');
```

Comment:
Should we put this code into a test helper class?

Also, there is a bunch of phpcs errors here

Reply:
I think it's only relevant for the Traversal test so I'm not sure if we'll ever re use it.